### DISCUSSION TITLE
Nuxt 3 dynamic routes doesn't work

### DISCUSSION THREAD

**Discussion by thewebartisan7** [nuxt/nuxt#18298] [Questions] [answered]
I try to setup dynamic route with nuxt 3 but it didn't work.
I can't find anything about it in docs, so I am referring to nuxt 2 docs, where I am using the underscore prefix.

So for example a simple folder structure like below doesn't work:

- pages/index.vue
- pages/posts/index.vue
- pages/posts/_slug/index.vue
- pages/posts/_slug/comments.vue

What I am missing?

https://stackblitz.com/edit/github-fnfktb-jyht6v?file=pages/posts/_slug/index.vue

![Screenshot-2021-10-19-at-08-59-09](https://user-images.githubusercontent.com/57477934/137859703-36dcab3b-e100-44fe-9f5d-68b5af112ef1.png)

**Comment by visini** [comment, marked as answer]
See here: http://v3.nuxtjs.org/docs/directory-structure/pages#dynamic-routes

```
-| pages/
---| index.vue
---| users/
-----| [id].vue
```

Or catchall:

```
-| pages/
---| index.vue
---| posts/
-----| [...slug].vue
```

**Reply by thewebartisan7** [reply]
Ah Thank you! I carefully checked all docs but not sure how but I skipped this part.
